.github/workflows/delete_backport_branch.yml

```diff
@@ -7,9 +7,16 @@ on:
 jobs:
   delete-branch:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
```

Comment:
I'm pretty sure this won't work with the pull request trigger, I think it needs to be updated to pull_request_target.

Reply:
Did not change the trigger. It is what previously existed. Pretty sure it works as we want to run this once a pull request in merged (type: closed) in this repository.

Reply:
I'm not so sure, otherwise it would seem that a bad actor change the branch target in a PR and delete main or 2.x on us.  Maybe this works because of the permissions of the bot account or your maintainer account?

Reply:
The already existing `if` here https://github.com/opensearch-project/OpenSearch/pull/15983/files#diff-0153e43e8683726e844f93b477ceed014452fc9006230d868bb6210bbff54d2aR12 would be the gatekeeper right?

Reply:
On a PR the author could remove that check, the PR author shouldn't be a gatekeeper for security concerns.  But you know what, either it will work or it won't - lets try this out and if its broken it won't be any worse than before.

Reply:
Also noticed one thing. We have branch protection rules in place that do not allow branch deletions by default. Only the branches with `backport/` are allowed to be deleted.